ui-manchette/src/types.ts

```diff
@@ -10,7 +10,14 @@ export type Waypoint = {
 export type InteractiveWaypoint = Waypoint & {
   styles?: CSSProperties;
   display?: boolean;
-  onClick?: (opId: string, opRef: HTMLDivElement | null) => void;
+  onClick?: (waypointId: string, waypointRef: React.RefObject<HTMLDivElement>) => void;
+};
+
+export type WaypointMenuData = {
+  menu: React.ReactNode;
+  activeWaypointId?: string;
+  activeWaypointRef?: React.RefObject<HTMLDivElement>;
```

Comment:
It's a bit cumbersome for the library user to grab the active waypoint ref from the `onClick` callback, then pass it back as a prop to the library. It would be nicer if this ref was entirely managed by the library itself.

To do so, we can make it so `Manchette` creates the ref, and passes it down to the active `Waypoint`. Basically,  we can:

- In `Waypoint`, add a new `waypointRef?: React.Ref<HTMLDivElement>` property
- In `WaypointList`, add a new `activeWaypointRef?: React.Ref<HTMLDivElement>` property and pass `waypointRef={activeWaypointId === waypoint.id ? activeWaypointRef : null}`
- In `Manchette`, create the `activeWaypointRef` ref and pass it to `WaypointList`

And then we can drop `activeWaypointRef` from the story/callback/etc.